docs/DeveloperGuide.md

```diff
@@ -4,30 +4,148 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+* [JSON-java](https://github.com/stleary/JSON-java)
+* [three-ten-extra](https://www.threeten.org/threeten-extra/)
+* * Requesting and Parsing of data from API into Java
+* https://www.youtube.com/watch?v=lDEfoSwyYFg
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design
+![](diagrams/Overall.png)
+Our main `Duke` class is responsible for the instantiation and launch of our application.  
+Our overall project design is split into 5 components, `command`, `common`,`data`, `storage` and `parser`.
+ - `command`: The command executor.
+ - `data`: Holds data and data structures of our application.
+ - `parser`: Reads data from the user input and parses it into an 'executable' command.
+ - `common`: Holds mainly static data that is used by multiple components.
+ - `storage`: Handles the reading and storing of our data.
+
+## Implementation
+### 'Add' feature
+This mechanism is facilitated by `CommandAdd`, which extends `Command`, as well as `ParserAdd`. `ParserAdd` parses the 
+user input into a string of words `parsedInput[]` which `CommandAdd` then stores internally. `CommandAdd` also stores an 
+internal reference to the list of expenses `expenseList` to be added to.
+
+`CommandAdd` implements the following operations:
+ - `CommandAdd#execute()` -- Instantiates the `Expense` object with the parsed inputs and adds it to `expenseList`
+ - `CommandAdd#roundInput(amount)` -- Rounds the expense amount to 2 decimal places. 
+
+`ParserAdd` implements the following operations:
+ - `ParserAdd#parseInput(userInput)` -- Parses the user input into an array of strings for `CommandAdd` to read.
+ - `ParserAdd#checkType(input)` -- Checks the type of the input corresponding to its index.
+ - `ParserAdd#substringIndex(type)` -- Returns the substring index of the start of the user input.
+
+The `CommandAdd#execute()` operation is exposed in the main `Duke` class, while the operations in `ParserAdd` are
+exposed in the `Parser` class.
+
+Given below is an example usage of the feature.
+![](./diagrams/AddFeature.png)
```

Comment:
Perhaps it would be nice to standardize the fonts in the diagrams, as the words in some diagrams are bolded and the words in others are not bolded